CONTRIBUTING.md

```diff
@@ -95,6 +96,98 @@ have more control over the process and debug any issues.
+      | where Timestamp > beginTime
+      | where LogId == lintDiffLogId
+      | where Message startswith_cs lintDiffLogMessageStart
+      | extend LintDiffViolations = parse_json(Message)
+      | mv-expand LintDiffViolations
+      | extend ViolationCode = tostring(LintDiffViolations.code)
+      | where ViolationCode == ruleName
+      )
+      on BuildId
+    | summarize count() by Time=bin(Timestamp, violationTimeBin), ViolationCode, PullRequestLink, BuildLink
+    | sort by count_ desc
+    ```
+    This will give you a list of the builds where the spec violated your rule and the count of violations for that build.
+    It includes a link to the PR as well. Using this list, visit the build page, click on LintDiff to view the logs, and
+    see where the rule was violated, you can then find that line in the spec by viewing the PR.
+    1. Wait until your rule has run on at least 10 different PRs and you have verified there are no false positives
+    1. It might also be helpful to view more than only the LintDiff results. You can use this Kusto query to see violations
+    of your new rule for an API spec without taking into account the previous version of the spec:
+    ```kusto
+    // edit these as desired
+    let beginTime = ago(7d);
+    let violationTimeBin = 10m;
+    let includeStagingPipeline = true;
+    let includeProdPipeline = false;
+    let ruleName = "ReservedResourceNamesModelAsEnum";
+    // do not edit these
+    let prodPipelineName = "Azure.azure-rest-api-specs-pipeline";
+    let stagingPipelineName = "Azure.azure-rest-api-specs-pipeline-staging";
+    let lintDiffLogId = 62;
+    let lintDiffLogMessageStart = "          \"code\":";
+    Build
+    | where StartTime > beginTime
+    | extend PipelineName = parse_json(Data).definition.name
+    | where PipelineName in (iff(includeStagingPipeline, stagingPipelineName, prodPipelineName), iff(includeProdPipeline, prodPipelineName, stagingPipelineName))
+    | extend SplitBranch = split(SourceBranch, "/")
+    | extend PullRequestLink = strcat("https://github.com/", SplitBranch[3], "/", SplitBranch[4], "/pull/", SplitBranch[5])
+    | extend BuildLink = tostring(parse_json(Data)._links.web.href)
+    | project StartTime, BuildId, SourceBranch, SourceVersion, PipelineName, PullRequestLink, BuildLink
+    | join kind=inner 
+      (
+      BuildLogLine
+      | where Timestamp > beginTime
+      | where LogId == lintDiffLogId
+      | where Message startswith_cs lintDiffLogMessageStart
+      | extend ViolationCode=trim_end('",', substring(Message, indexof(Message, ': "')+3))
+      | where ViolationCode == ruleName
+      )
+      on BuildId
+    | summarize count() by Time=bin(Timestamp, violationTimeBin), ViolationCode, PullRequestLink, BuildLink
+    | sort by count_ desc 
+    ```
+1. Once you verify the rules work correctly, roll them out to the production pipeline by undoing the staging-only setting
+from [Deploy to Staging LintDiff](#deploy-to-staging-lintdiff).
```

Comment:
This goes to previous line.

Reply:
Markdown will render this all as one line. I truncate the line length to ~120 characters as the markdown lint VS code extension suggests doing it and it helps keep the lines readable on smaller screens without changing the resulting page.